packages/react/src/hooks/useAuth.ts

```diff
@@ -1,38 +1,20 @@
 import type { ActJWTClaim, GetToken, SignOut } from '@clerk/types';
+import { useMemo } from 'react';
 
 import { useAuthContext } from '../contexts/AuthContext';
 import { useIsomorphicClerkContext } from '../contexts/IsomorphicClerkContext';
 import { invalidStateError } from '../errors';
 import { createGetToken, createSignOut } from './utils';
 
-type UseAuthReturn =
-  | {
-      isLoaded: false;
-      isSignedIn: undefined;
-      userId: undefined;
-      sessionId: undefined;
-      actor: null;
-      signOut: SignOut;
-      getToken: GetToken;
-    }
-  | {
-      isLoaded: true;
-      isSignedIn: false;
-      userId: null;
-      sessionId: null;
-      actor: null;
-      signOut: SignOut;
-      getToken: GetToken;
-    }
-  | {
-      isLoaded: true;
-      isSignedIn: true;
-      userId: string;
-      sessionId: string;
-      actor: ActJWTClaim | null;
-      signOut: SignOut;
-      getToken: GetToken;
-    };
+type UseAuthReturn = {
```

Comment:
Can you please elaborate on the reasoning for this change? This new type definition isn't the same as the previous one. The main reason is that by declaring a union of type objects we create a stricter `UseAuthReturn` type that allows only the declared combinations.

Reply:
> PS, modified the types a bit, unfortunately I'm not that familiar with TS but the default installation was giving me types issues [...] the one modifying UseAuthReturn was because I had trouble with getting the highly-dynamic variable to work otherwise.

If I change just the scope of `getToken()` (+ `signOut()`) I believe I might be able to bring the original types back.

Side question: why is it so... messy? IMHO `isSignedIn` should be `true` or `false`, no chance for `undefined`. Same with `userId` or `sessionId`, why not make them either `string | null`? Right now the potential combinatorial explosion is scary and you _need_ the types to make sense of it, but if simplified well it'd be much better (again just my opinion/tip, this is your project after all).